File: LeetCode-December-Challenge/DetermineifStringHalvesAreAlike.java
class Solution {
    private boolean isVowel(char s){
        return s == 'a' || s == 'e' || s == 'i' || s == 'o' || s == 'u' || s == 'A' || s == 'E' || s == 'O' || s == 'U' || s == 'I';
    }
    public boolean halvesAreAlike(String s) {
        int half = 0;
        for(int i = 0; i < s.length() / 2; ++i) if(isVowel(s.charAt(i))) half++;
        for(int i = s.length() / 2; i < s.length(); ++i) if(isVowel(s.charAt(i))) half--;
        return half == 0;
    }
}
public class DetermineifStringHalvesAreAlike {
    public static void main(String[] args) {
        String s = "book";
        Solution S = new Solution();
        System.out.println(S.halvesAreAlike(s));
    }
}

File: LeetCode-December-Challenge/DetermineifStringHalvesAreAlike.js
var halvesAreAlike = function(s) {
    let vowels = ['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'];
    let half = 0;
    for(let i = 0; i < s.length / 2; ++i) if(vowels.indexOf(s[i]) > -1)half++;
    for(let i = s.length / 2; i < s.length; ++i) if(vowels.indexOf(s[i]) > -1)half--;
    return half === 0;
};
console.log(halvesAreAlike('textbook'));